odra-casper/livenet/src/casper_client.rs

```diff
@@ -0,0 +1,445 @@
+            } => {
+                panic!(
+                    "Deploy {:?} failed with error: {:?}",
+                    deploy_hash_str, error_message
+                );
+            }
+            casper_types::ExecutionResult::Success {
+                effect: _,
+                transfers: _,
+                cost: _
+            } => {
+                println!("Deploy {:?} successfully executed", deploy_hash_str);
+            }
+        }
+    }
+
+    fn new_deploy(&self, session: ExecutableDeployItem, gas: Balance) -> Deploy {
+        let timestamp = Timestamp::now();
+        let ttl = TimeDiff::from(10000000);
+        let gas_price = 1;
+        let dependencies = vec![];
+        let chain_name = String::from(self.chain_name());
+        let payment = ExecutableDeployItem::ModuleBytes {
+            module_bytes: Default::default(),
+            args: runtime_args! {
+                "amount" => gas
+            }
+        };
+
+        Deploy::new(
+            timestamp,
+            ttl,
+            gas_price,
+            dependencies,
+            chain_name,
+            payment,
+            session,
+            &self.secret_key,
+            Some(self.public_key())
+        )
+    }
+
+    fn post_request<T: DeserializeOwned>(&self, request: Value) -> T {
+        let client = reqwest::blocking::Client::new();
+        let response = client
+            .post(self.node_address_rpc())
+            .json(&request)
+            .send()
+            .unwrap();
+        let response: JsonRpc = response.json().unwrap();
+        let response = response.get_result().unwrap();
+        serde_json::from_value(response.clone()).unwrap()
+    }
+}
+
+fn find_wasm_file_path(wasm_file_name: &str) -> PathBuf {
+    let mut path = PathBuf::from("wasm").join(wasm_file_name);
+    let mut checked_paths = vec![];
+    for _ in 0..2 {
+        if path.exists() {
+            println!("Found wasm under {:?}", path);
+            return path;
+        } else {
+            checked_paths.push(path.clone());
+            path = path.parent().unwrap().to_path_buf();
+        }
+    }
+    panic!("Could not find wasm under {:?}", checked_paths);
+}
+
+struct LivenetKeyMaker;
+
+impl KeyMaker for LivenetKeyMaker {
+    fn blake2b(preimage: &[u8]) -> [u8; 32] {
+        let mut result = [0; 32];
+        let mut hasher = VarBlake2b::new(32).expect("should create hasher");
+
+        hasher.update(preimage);
+        hasher.finalize_variable(|slice| {
+            result.copy_from_slice(slice);
+        });
+        result
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::str::FromStr;
+
+    use casper_hashing::Digest;
+    use casper_node::types::DeployHash;
+    use casper_types::{
+        bytesrepr::{FromBytes, ToBytes},
+        U256
+    };
+    use odra_casper_types::Address;
+
+    use super::CasperClient;
+
+    const CONTRACT_PACKAGE_HASH: &str =
+        "hash-40dd2fef4e994d2b0d3d415ce515446d7a1e389d2e6fc7c51319a70acf6f42d0";
```

Comment:
Those values will be different depending on ENV_CHAIN_NAME, they should be also put in the env file, or the node and chain name also hardcoded for testing.